### PR TITLE
MODPUBSUB-40 - Increase test coverage for mod-pub-sub

### PR DESCRIPTION
**Pull request by RuslanLavrov**

### CODE REVIEW

**Comment by OleksiiKuzminov** [comment]
plz return save\delete methods and add unit tests

File: mod-pubsub-server/src/main/java/org/folio/dao/MessagingModuleDao.java [hunk 1]
```diff
@@ -37,23 +28,6 @@
...
   Future<List<MessagingModule>> save(String moduleName, List<MessagingModule> messagingModules);
 
-  /**
-   * Updates {@link MessagingModule} in data base
-   *
-   * @param id messagingModule id
-   * @param messagingModule entity to update
-   * @return future with updated MessagingModule
-   */
-  Future<MessagingModule> update(String id, MessagingModule messagingModule);
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/MessagingModuleDao.java]: -

File: mod-pubsub-server/src/main/java/org/folio/dao/MessagingModuleDao.java [hunk 2]
```diff
@@ -37,23 +28,6 @@
...
-
-  /**
-   * Deletes {@link MessagingModule} by id
-   *
-   * @param id messagingModule id
-   * @return future with boolean
-   */
-  Future<Boolean> delete(String id);
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/MessagingModuleDao.java]: +

File: mod-pubsub-server/src/main/java/org/folio/dao/ModuleDao.java [hunk 3]
```diff
@@ -5,29 +5,13 @@
...
  * Module entity data access object
  */
 public interface ModuleDao {
 
-  /**
-   * Searches for all {@link Module} in database
-   *
-   * @return return future with Module list
-   */
-  Future<List<Module>> getAll();
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/ModuleDao.java]: +

File: mod-pubsub-server/src/main/java/org/folio/dao/ModuleDao.java [hunk 4]
```diff
@@ -37,14 +21,6 @@
    */
   Future<Optional<Module>> getByName(String name, AsyncResult<SQLConnection> sqlConnection);
 
-  /**
-   * Saves new {@link Module} to data base
-   *
-   * @param module Module entity to save
-   * @return module id
-   */
-  Future<String> save(Module module);
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/ModuleDao.java]: +

File: mod-pubsub-server/src/main/java/org/folio/dao/ModuleDao.java [hunk 5]
```diff
@@ -53,21 +29,4 @@
...
-   */
-  Future<Module> update(String id, Module module);
-
-  /**
-   * Deletes {@link Module} by id
-   *
-   * @param id Module id
-   * @return future with boolean
-   */
-  Future<Boolean> delete(String id);
```
Comment (OleksiiKuzminov) [inline mod-pubsub-server/src/main/java/org/folio/dao/ModuleDao.java]: +

File: mod-pubsub-server/src/main/java/org/folio/services/impl/MessagingModuleServiceImpl.java [hunk 6]
```diff
@@ -115,7 +118,24 @@ private void compareEventDescriptors(EventDescriptor eventDescriptor, EventDescr
       .collect(Collectors.toMap(SubscriptionDefinition::getEventType, SubscriptionDefinition::getCallbackAddress));
     messagingModules.forEach(module -> module.setSubscriberCallback(subscriberCallbacksMap.get(module.getEventType())));
 
-    return messagingModuleDao.save(subscriberDescriptor.getModuleName(), messagingModules).map(true);
+    return clearPreviousMessagingModulesInfo(subscriberDescriptor.getModuleName(), SUBSCRIBER, tenantId)
+      .compose(deleted ->  messagingModuleDao.save(subscriberDescriptor.getModuleName(), messagingModules))
```
Comment (Igor-Gorchakov) [inline mod-pubsub-server/src/main/java/org/folio/services/impl/MessagingModuleServiceImpl.java]: In one transaction ?

**Pull request merged**